diff --git a/preupg/report_parser.py b/preupg/report_parser.py
index 1a9047c..7223bf4 100644
--- a/preupg/report_parser.py
+++ b/preupg/report_parser.py
@@ -198,6 +198,25 @@ class ReportParser(object):
 
         self.write_xml()
 
+    def fix_stderr(self):
+        """The function transform stdout to stderr."""
+        for rule in self.get_all_result_rules():
+        # Filter all check-import=stdout which are empty and remove them
+            for check in self.get_nodes(rule, "check"):
+                cim_stdout = [x for x in self.get_nodes(check, "check-import")
+                                if x.get('import-name') == "stdout"]
+                cim_stderr = [x for x in self.get_nodes(check, "check-import")
+                                if x.get('import-name') == "stderr"]
+                if not cim_stdout or cim_stdout[0].text is None or cim_stdout[0].text == "":
+                    continue
+
+                # unexpected, but just to be sure..
+                if cim_stderr:
+                    if cim_stderr[0].text is not None and cim_stderr[0].text != "":
+                        cim_stdout[0].text +="\n"+cim_stderr[0].text
+                    remove_node(check, cim_stderr[0])
+                cim_stdout[0].set('import-name', "stderr")
+
     def update_inplace_risk(self, scanning_progress, rule, res):
         """Function updates inplace risk"""
         inplace_risk = XccdfHelper.get_check_import_inplace_risk(rule)
@@ -222,6 +241,7 @@ class ReportParser(object):
         """
         #Filter all rule-result in TestResult
         changed_fields = []
+        self.fix_stderr() # it is ugly but command below too
         self.remove_empty_check_import()
         inplace_dict = {
             0: ReportHelper.upd_inspection,
